### PR TITLE
[gardening][enum class] Replace `unsigned char` with `uint8_t` for consistency

### PR DESCRIPTION
**Pull request by ikesyo**
Before the changes:

- `git grep -E "enum class .+ : uint8_t \{" | wc -l`: 90
- `git grep -E "enum class .+ : unsigned char \{" | wc -l`: 26

### CODE REVIEW

File: lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp [hunk 1]
```diff
@@ -191,11 +191,7 @@ static void InsertCFGDiamond(SILValue Cond, SILLocation Loc, SILBuilder &B,
 //===----------------------------------------------------------------------===//
 
 namespace {
-  enum class DIKind : unsigned char {
-    No,
-    Yes,
-    Partial
-  };
+enum class DIKind : uint8_t { No, Yes, Partial };
```
Comment (ikesyo) [inline lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp]: The format change is due to `git clang-format`.

**Comment by ikesyo** [comment]
@swift-ci smoke test

**Review by compnerd — APPROVED** [review]
This is probably nicer to do.

**Comment by DougGregor** [comment]
Yeah, it's good to be explicit.

**Pull request merged**